pkg/ui/react-app/src/thanos/pages/blocks/Blocks.tsx

```diff
@@ -21,10 +21,27 @@ export interface BlockListProps {
 
 export const BlocksContent: FC<{ data: BlockListProps }> = ({ data }) => {
   const [selectedBlock, selectBlock] = useState<Block>();
-
-  const { blocks, label, err } = data;
+  const linkRef: any = useRef();
+  const [ready, setReady] = useState(false);
+  const [open, setOpen] = useState(false);
+  const toggle = () => setOpen(!open);
```

Comment:
```suggestion
  const toggle = () => setOpen((o) => !o);
```

The state setter functions are async in nature, using it like a closure (accessing the state from outside) can result in stale updates/ race conditions. Take a look at https://reactjs.org/docs/hooks-reference.html#functional-updates